### DISCUSSION TITLE
Is there any way to know if RSI is uptrend or downtrend?

### DISCUSSION THREAD

**Discussion by prasad-zenconix** [DaveSkender/Stock.Indicators#900] [Q&A] [answered]
Hi there, 
I am looking for some solution or hint to get if RSI is uptrend or downtrend.

if it is continues uptrend like 40 -> 45 -> 50 ->  55 ->  58 ->  60 then its easy to figure out , it is uptrend 
but if it is something like 40 -> 45 -> 50 -> 55 -> 53 -> 55 -> 58 -> 56 -> 60 ... although overall it is uptrend but little bit difficult to know, that it is uptrend.. 
Same for downtrend .. 
Is there any way to know if it is going uptrend or downtrend ? any hint is ok. 

Thank you.

**Comment by DaveSkender** [comment, marked as answer]
There's probably a few concepts you can try.  Here's some random thoughts:

```csharp
// average directional index
var results = quotes
  .GetRsi(..)
  .GetAdx(..);

// slope of smoothed RSI
var results = quotes
  .GetRsi(..)
  .GetAlma(..)
  .GetSlope(..);
```

**Reply by DaveSkender** [reply]
You can also try an STC of an RSI.  I bet you can get it to give you a binary-like output if you get the parameter values right:

```csharp
var results = quotes
  .GetRsi(..)
  .GetStc(..);
```

**Reply by prasad-zenconix** [reply]
Thank you DaveSkendar for help.

**Reply by DaveSkender** [reply]
This might work too:

```csharp
var results = quotes
  .GetRsi(..)
  .GetAroon(..);
```

**Reply by prasad-zenconix** [reply]
Thank you DaveSkendar , I will try with Aroon. Thanks a lot for help
